flowforge/input/Components.py

```diff
@@ -309,6 +309,148 @@ def factory(indict: Dict) -> Dict[str, Component]:
         return components
 
 
+class CrossSection(abc.ABC):
+    """Abstract base class for cross-sectional areas"""
+
+    @abc.abstractmethod
+    def flow_area(self) -> float:
+        pass
+
+    @abc.abstractmethod
+    def hydraulic_diameter(self) -> float:
+        pass
+
+    @abc.abstractmethod
+    def wetted_perimeter(self) -> float:
+        pass
+
+class CircularCrossSection(CrossSection):
+    """Circular cross-sectional area
+
+    Parameters
+    ---------
+    R: float
+        Radius of the circular pipe
+    """
+
+    def __init__(self, R: float):
+        self._R = R
+
+    @property
+    def R(self) -> float:
+        return self._R
+
+    def flow_area(self) -> float:
+        return np.pi * self._R ** 2
+
+    def hydraulic_diameter(self) -> float:
+        return 2 * self._R
+
+    def wetted_perimeter(self) -> float:
+        return 2 * np.pi * self._R
+
+
+class SquareCrossSection(CrossSection):
```

Comment:
I would make `SquareCrossSection` a subclass of `RectangularCrossSection`.  All you'll have to write then is the `__init__` and the rest will just be inherited